static/js/components/HomePage.jsx.template

```diff
@@ -0,0 +1,197 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import { Responsive, WidthProvider } from "react-grid-layout";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import "reactgridlayoutcss/styles.css";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import "reactresizablecss/styles.css";
+
+import { makeStyles } from "@material-ui/core/styles";
+import IconButton from "@material-ui/core/IconButton";
+import CachedIcon from "@material-ui/icons/Cached";
+import Tooltip from "@material-ui/core/Tooltip";
+
+import * as profileActions from "../ducks/profile";
+
+import RecentSources from "./RecentSources";
+import GroupList from "./GroupList";
+import NewsFeed from "./NewsFeed";
+import TopSources from "./TopSources";
+import SourceCounts from "./SourceCounts";
+import UninitializedDBMessage from "./UninitializedDBMessage";
+
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+const useStyles = makeStyles(() => ({
+  widgetIcon: {
+    float: "right",
+    color: "gray",
+    margin: "0.25rem 0.25rem 0.25rem 0.25rem",
+    "&:hover": {
+      cursor: "pointer",
+    },
+  },
+  widgetPaperDiv: {
+    padding: "1rem",
+    height: "100%",
+  },
+  widgetPaperFillSpace: {
+    height: "100%",
+  },
+  resetButton: {
+    position: "relative",
+    float: "right",
+    margin: "0.5rem",
+  },
+}));
+
+{%- set ns = namespace(xlg=[], lg=[], md=[], sm=[], xs=[]) -%}
+{%- for widgetName, widgetProps in app.homepage_widgets.items() %}
+  {%- if widgetProps.show %}
+    {%- set ns.xlg = [{
```

Comment:
Use a for loop to generate repeating items like this.

Reply:
@stefanv I tried for awhile yesterday to get it to work with for loops but couldn't get past Jinja limitations. Specifically, I couldn't figure out a way to access namespace variables via variable keys, i.e.
![Screenshot 2020-09-03 183740](https://user-images.githubusercontent.com/17696889/92189866-9254a980-ee14-11ea-8fbd-4ec83ecc515f.png)
^ and variations from that didn't work at all because Jinja didn't like the bracket syntax at all on the namespace for some reason

If you have any suggestions on how to use the for loops, that'd be great.